### ISSUE TITLE
Typing seems incorrect for useAnimatedRef + FlatList + scrollToIndex

### ISSUE THREAD

**Issue by Jpunt** [software-mansion/react-native-reanimated#4616]
### Description

First of all, I am aware of the fact that `scrollTo` exists (because https://github.com/software-mansion/react-native-reanimated/issues/2005 led me to it), but for some reason this does nothing in my situation. The thing to use next would be a `useAnimatedRef` in combination with `scrollToIndex` which works fine, but the typing is incorrect. 

### Steps to reproduce

1.   Create a ref:
```ts
const listRef = useAnimatedRef<Animated.FlatList<T>>()
```

2. Create a flatlist:
```tsx
<Animated.FlatList
  ref={listRef}
  data={data}
  // ...
/>
```

3. Create a function to scroll to the top:
```ts
// TS-Error: scrollToIndex does not exists
// Works runtime
listRef.current?.scrollToIndex({ index: 0, animated: true }) 

// No TS-error
// Runtime error: undefined is not a function
listRef.current?.getNode().scrollToIndex({ index: 0, animated: true }) 
```

### Snack or a link to a repository

Will add this later if needed

### Reanimated version

2.17.0

### React Native version

0.71.6

### Platforms

Android, iOS

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

None

### Device

None

### Acknowledgements

Yes

**Comment by jobpaardekooper** [comment]
Can confirm that I am also having this exact issue. This might get fixed by #4519. Maybe @tjzel can confirm?

**Comment by tjzel** [comment]
Hi @Jpunt. The issue you are referring to is known to me. Amidst many errors or inaccuracies in `react-native-reanimated.d.ts` is accidental declaration of `Animated.FlatList` function component as `View` from `react-native`.

I had found it some time ago and initially wanted to fix it with all the other mistakes but this was an extremely difficult task and I decided not to correct it at all and just drop the file.

It's still going to be some time before #4519 gets included into next release so if it's really bothering you and you want to keep your TypeScript code clean all you need to do is change the following line in `react-native-reanimated.d.ts`:

```Diff
-    export interface FlatList<T> extends ReactNativeView<T> {}
+    export interface FlatList<T> extends ReactNativeFlatList<T> {}
```

I'll keep the issue open until required fix gets merged.
